src/eventListeners/onImageRenderedBrushEventHandler.js

```diff
@@ -54,27 +63,29 @@ function shouldRenderSegmentation (evt, segIndex, segData) {
   const toolData = segData.toolData;
   const visibleSegmentations = segData.visibleSegmentations;
 
-  if (!toolData.data[segIndex].pixelData) { // No data, no render.
+  if (!toolData.data[segIndex].pixelData) {
+    // No data, no render.
     return false;
   }
 
-  if (visibleSegmentations[segIndex]) { // Has data and marked as visible, render!
+  if (visibleSegmentations[segIndex]) {
+    // Has data and marked as visible, render!
     return true;
   }
 
-  const currentColor = brushState.getters.draw();
+  const currentColor = state.drawColorId;
```

Comment:
I'm a bit torn on this. Part of me feels like we should call a getter/setter for any interaction with state, but internally there isn't any problem with accessing the state directly. I think we can access the state directly from inside `cornerstoneTools`, but should definitely make sure everything that may changed via an external call has get/set api endpoints

We should probably wrap this in getters, no? I think anything that could be public API (change color surely is, should be changed through a api method we can easily doc, rather than modifying

Reply:
> should definitely make sure everything that may changed via an external call has get/set api endpoints

That makes sense to me. Anything that we want to expose as an external API call should have:

- JS Doc comments
- Be unit tested (eventually... haha)
- Use getters/setters

What's a little difficult is that the entire store is a "Public API" from the perspective of 3rd party tool authors. 🤔 

So should `state` only be `private`, and used exclusively by `getters`/`setters`? FWIW, Vue uses:

- state: can be imported and used by anything, but only updated by `mutations`
- getters: for computed/formatted/filtered/mapped values from `state`
- mutations: the only way to change a state property's value

So some a given module might use `state` or `getters` to retrieve a value, but only `mutations` to modify.